src/cli/npc.rs

```diff
@@ -99,6 +106,29 @@ fn _is_route_to_specified_dev(route: &Route, iface_name: &str) -> bool {
     false
 }
 
+fn get_route_scope_from_str(route_scope_str: &str) -> RouteScope {
+    if route_scope_str == ROUTE_SCOPE_STRINGS[1]
+        || route_scope_str == ROUTE_SCOPE_STRINGS[2]
+        || route_scope_str == ROUTE_SCOPE_STRINGS[3]
+    {
+        RouteScope::Universe
+    } else if route_scope_str == ROUTE_SCOPE_STRINGS[4]
+        || route_scope_str == ROUTE_SCOPE_STRINGS[5]
+    {
+        RouteScope::Site
+    } else if route_scope_str == ROUTE_SCOPE_STRINGS[6]
+        || route_scope_str == ROUTE_SCOPE_STRINGS[7]
+    {
+        RouteScope::Link
+    } else if route_scope_str == ROUTE_SCOPE_STRINGS[8]
+        || route_scope_str == ROUTE_SCOPE_STRINGS[9]
```

Comment:
Hard coded magic number is vulnerable to human errors.

How about:

```rust
match route_scope_str {
    "universe" || "u" => RouteScope::Universe,
   _ => Err("Invalid route scope"),
}
```

And the filter line could be:

```rust
.filter(|route| scope == "all" || route.scope == get_route_scope_from_str(scope))
```

Reply:
Thanks for the suggestions! I followed them and I have added some code to handle the error case when the route scope is invalid, it returns a `CliError` if necessary.

I kept the `ROUTE_SCOPE_STRINGS` array so `clap` can validate the users' input beforehand.